### ISSUE TITLE
`setup.strategy: sync` does not sync sites

### ISSUE THREAD

**Issue by timwood** [acquia/blt#2559]
My system information:
* Operating system type: Mac OS
* Operating system version: Sierra
* BLT version: 8.9.13

Output of `blt doctor`:
```
+----------------------+----------------------------------------------------------------------------------+
| Property             | Value                                                                            |
+----------------------+----------------------------------------------------------------------------------+
| %paths               |                                                                                  |
|  - %root             | /var/www/site/docroot                                                        |
|  - %site             | sites/default                                                                    |
|  - %modules          | modules                                                                          |
|  - %themes           | themes                                                                           |
|  - %config-sync      | /var/www/site/config/default                                                 |
|  - %files            | sites/default/files                                                              |
|  - %private          | /var/www/site/files-private                                                  |
|  - %temp             | /tmp                                                                             |
| admin-theme          | seven                                                                            |
| blt-version          | 8.9.13                                                                           |
| bootstrap            | Successful                                                                       |
| config-sync          | /var/www/site/config/default                                                 |
| db-driver            | mysql                                                                            |
| db-hostname          | localhost                                                                        |
| db-name              | drupal                                                                           |
| db-password          | drupal                                                                           |
| db-port              | 3306                                                                             |
| db-status            | Connected                                                                        |
| db-username          | drupal                                                                           |
| dev-desktop-enabled  | false                                                                            |
| drupal-settings-file | sites/default/settings.php                                                       |
| drupal-version       | 8.4.4                                                                            |
| drupal-vm-enabled    | true                                                                             |
| drush-alias-files    | /var/www/site/docroot/../drush/site-aliases/aliases.drushrc.php              |
|                      | /var/www/site/docroot/../drush/site-aliases/example.acsf.aliases.drushrc.php |
|                      | /var/www/site/docroot/../drush/site-aliases/example.local.aliases.drushrc.ph |
|                      | p                                                                                |
|                      | /var/www/site/docroot/../drush/site-aliases/site.aliases.drushrc.php        |
| drush-conf           | /var/www/v/drush/drushrc.php                                              |
| drush-script         | /var/www/site/vendor/drush/drush/drush.php                                   |
| drush-temp           | /tmp                                                                             |
| drush-version        | 8.1.16                                                                           |
| files                | sites/default/files                                                              |
| install-profile      | lightning                                                                        |
| local-drushrc        | /var/www/site/docroot/sites/default/local.drushrc.php                        |
| local-settings       | /var/www/site/docroot/sites/default/settings/local.settings.php              |
| modules              | modules                                                                          |
| php-bin              | /usr/bin/php                                                                     |
| php-conf             | /etc/php/7.1/cli/php.ini                                                         |
| php-mysql            | /var/run/mysqld/mysqld.sock                                                      |
| php-os               | Linux                                                                            |
| private              | /var/www/site/files-private                                                  |
| root                 | /var/www/site/docroot                                                        |
| shell                |                                                                                  |
| site                 | sites/default                                                                    |
| temp                 | /tmp                                                                             |
| theme                | uswds_site                                                                      |
| themes               | themes                                                                           |
| uri                  | https://local.site.test                                                 |
| user                 |                                                                                  |
+----------------------+----------------------------------------------------------------------------------+
+--------------------------+-----------------------------------------------------------------+
| Check                    | Outcome                                                         |
+--------------------------+-----------------------------------------------------------------+
| checkDocrootExists       | Found docroot.                                                  |
| checkCoreExists          | Drupal core exists                                              |
| checkSettingsFile        | BLT settings are included in settings file.                     |
| checkLocalSettingsFile   | Found your local settings file.                                 |
| checkLocalDrushFile      | Found your local drush settings file.                           |
| checkUri                 | $options['uri'] is set.                                         |
| checkUriResponse         | Received a response from site https://local.site.test. |
| checkHttps               | The SSL certificate for your local site appears valid.          |
| checkFileSystem:%files   | Public files directory is writable.                             |
| checkFileSystem:%private | Private files directory is writable.                            |
| checkFileSystem:%temp    | Temporary files directory is writable.                          |
| checkDbConnection        | Connected to database.                                          |
| checkDrupalBootstrapped  | Bootstrapped Drupal via drush.                                  |
| checkDrupalInstalled     | Drupal is installed.                                            |
| checkCaching:page        | Drupal cache is disabled.                                       |
| checkCaching:css         | CSS preprocessing is disabled.                                  |
| checkCaching:js          | JS preprocessing is disabled.                                   |
| checkComposer:require    | acquia/blt is in composer.json's require object.                |
| checkBehat:exists        | Behat local settings file exists.                               |
| checkBehat:root          | Behat drupal_root is set correctly for Drupal VM.               |
| checkProjectYml:keys     | project.yml has no deprecated keys.                             |
| checkDrupalVm:alias      | drush.aliases.local exists your drush aliases file.             |
| checkDrupalVm            | Drupal VM is configured correctly.                              |
| checkPhpDateTimezone     | PHP setting for date.timezone is correctly set                  |
+--------------------------+-----------------------------------------------------------------+

```

When I run this command:
```
blt setup
```

I get the following output:
```
...
> setup:drupal:install
Executing command blt 'setup:drupal:install' --define=drush.alias=self inside of Drupal VM...
[warning] The xDebug extension is loaded. This will significantly decrease performance.
> internal:drupal:install
[Acquia\Blt\Robo\Tasks\DrushTask] Running /var/www/site/vendor/bin/drush site-install lightning install_configure_form.update_status_module='array(FALSE,FALSE)' install_configure_form.enable_update_status_module=NULL --site-name=Site.com --site-mail=no-reply@acquia.com --account-name=yE5kpR_jDU --account-mail=no-reply@acquia.com --locale=en --config-dir=../config/default --uri=default --yes --verbose in /var/www/site/docroot
Executing: mysql --defaults-extra-file=/tmp/drush_Ob16X8 --database=drupal --host=localhost --port=3306 --silent  < /tmp/drush_fpuia0 > /dev/null
You are about to DROP all tables in your 'drupal' database. Do you want to continue? (y/n): y
Sites directory /var/www/site/docroot/sites/default already exists - proceeding.                                                                                                             [notice]
Executing: mysql --defaults-extra-file=/tmp/drush_SIqcCq --database=drupal --host=localhost --port=3306 --silent  < /tmp/drush_zFi8Mh > /dev/null
Executing: mysql --defaults-extra-file=/tmp/drush_mrKYhI --database=drupal --host=localhost --port=3306 --silent  < /tmp/drush_9aP3rz
Executing: mysql --defaults-extra-file=/tmp/drush_IorxZZ --database=drupal --host=localhost --port=3306 --silent  < /tmp/drush_lmdK8Q
Starting Drupal installation. This takes a while. Consider using the --notify global option.                                                                                                  [ok]
...
```

And I expected this to happen:
> sync
> sync:db
> setup:settings
[Acquia\Blt\Robo\Tasks\DrushTask] Running /Users/tww1/Sites/site/vendor/bin/drush cache-clear drush --uri=default --yes &&
 /Users/tww1/Sites/site/vendor/bin/drush sql-drop --uri=default --yes &&
 /Users/tww1/Sites/site/vendor/bin/drush sql-sync @site.dev @site.local --structure-tables-key=lightweight --create-db --sanitize --uri=default --yes &&
 /Users/tww1/Sites/site/vendor/bin/drush cache-clear drush --uri=default --yes in /Users/tww1/Sites/site/docroot

Just wondering whether adding the following to `blt/project.yml` or even `blt/project.local.yml` should make `blt setup` run a sync from the remote rather than running the Drupal site install?
```
setup:
  # Valid values are install, sync, import.
  strategy: sync
```

**Comment by grasmash** [comment]
Hm... setting `setup.strategy` to `sync` should work. Note that a value in `project.local.yml` will take precedence over any value in `project.yml`.

See https://github.com/acquia/blt/blob/8.9.x/src/Robo/Commands/Setup/AllCommand.php#L34

What is the output of `blt config:dump` ?

**Comment by timwood** [comment]
Output of `blt config:dump`. I've replaced site name throughout as I did before.

```
+----------------------------------------+----------------------------------------------------------------------------------+
| Property                               | Value                                                                            |
+----------------------------------------+----------------------------------------------------------------------------------+
| repo.root                              | /Users/me/Sites/site                                                       |
| docroot                                | /Users/me/Sites/site/docroot                                               |
| blt.root                               | /Users/me/Sites/site/vendor/acquia/blt                                     |
| blt.update.ignore-existing-file        | /Users/me/Sites/site/vendor/acquia/blt/scripts/blt/ignore-existing.txt     |
| blt.config-files.project               | /Users/me/Sites/site/blt/project.yml                                       |
| blt.config-files.local                 | /Users/me/Sites/site/blt/project.local.yml                                 |
| blt.config-files.example-local         | /Users/me/Sites/site/blt/example.project.local.yml                         |
| blt.config-files.schema-version        | /Users/me/Sites/site/blt/.schema_version                                   |
| blt.config-files.multisite             | /Users/me/Sites/site/docroot/sites/default/site.yml                        |
| blt.command-cache-dir                  | /Users/me/Sites/site/vendor/acquia/blt/cache/commands                      |
| composer.bin                           | /Users/me/Sites/site/vendor/bin                                            |
| composer.extra                         |                                                                                  |
| behat.config                           | /Users/me/Sites/site/tests/behat/local.yml                                 |
| behat.profile                          | local                                                                            |
| behat.selenium.port                    | 4444                                                                             |
| behat.selenium.url                     | http://127.0.0.1:4444/wd/hub                                                     |
| behat.chrome.port                      | 9222                                                                             |
| behat.chrome.args                      |                                                                                  |
| behat.paths.0                          | /Users/me/Sites/site/tests/behat                                           |
| behat.tags                             | ~ajax&&~experimental&&~lightningextension                                        |
| behat.extra                            |                                                                                  |
| behat.web-driver                       | chrome                                                                           |
| bin.path                               | vendor/bin                                                                       |
| cm.strategy                            | config-split                                                                     |
| cm.core.path                           | ../config                                                                        |
| cm.core.key                            | sync                                                                             |
| cm.core.dirs.sync.path                 | ../config/default                                                                |
| cm.features.no-overrides               | 1                                                                                |
| deploy.build-dependencies              | 1                                                                                |
| deploy.dir                             | /Users/me/Sites/site/deploy                                                |
| deploy.exclude_file                    | /Users/me/Sites/site/vendor/acquia/blt/scripts/blt/deploy/deploy-exclude.t |
|                                        | xt                                                                               |
| deploy.exclude_additions_file          | /Users/me/Sites/site/blt/deploy-exclude-additions.txt                      |
| deploy.gitignore_file                  | /Users/me/Sites/site/vendor/acquia/blt/scripts/blt/deploy/.gitignore       |
| deploy.git.failOnDirty                 | 1                                                                                |
| drupal.account.mail                    | no-reply@acquia.com                                                              |
| drupal.locale                          | en                                                                               |
| drupal.local_settings_file             | /Users/me/Sites/site/docroot/sites/default/settings/local.settings.php     |
| drupal.settings_file                   | /Users/me/Sites/site/docroot/sites/default/settings.php                    |
| drush.bin                              | /Users/me/Sites/site/vendor/bin/drush                                      |
| drush.dir                              | /Users/me/Sites/site/docroot                                               |
| drush.sanitize                         | 1                                                                                |
| drush.aliases.remote                   | site.dev                                                                        |
| drush.aliases.local                    | site.local                                                                   |
| drush.aliases.ci                       | self                                                                             |
| drush.default_alias                    | site.local                                                                   |
| drush.alias                            | site.local                                                                   |
| drush.uri                              | default                                                                          |
| git.hooks.pre-commit                   | /Users/me/Sites/site/vendor/acquia/blt/scripts/git-hooks                   |
| git.hooks.commit-msg                   | /Users/me/Sites/site/vendor/acquia/blt/scripts/git-hooks                   |
| git.default_branch                     | master                                                                           |
| git.remotes.0                          | site@svn-XXXXX.prod.hosting.acquia.com:site.git                                |
| reports.localDir                       | /Users/me/Sites/site/reports                                               |
| reports.remoteDir                      | reports                                                                          |
| phpcs.standard                         | Drupal,DrupalPractice                                                            |
| phpcbf.filesets.0                      | files.php.custom.modules                                                         |
| phpcbf.filesets.1                      | files.php.tests                                                                  |
| phpcbf.filesets.2                      | files.php.custom.themes                                                          |
| phpcbf.filesets.3                      | files.frontend.custom.themes                                                     |
| phpunit.0.path                         | /Users/me/Sites/site/tests/phpunit                                         |
| phpunit.0.config                       |                                                                                  |
| project.local.uri                      | https://local.site.test                                                 |
| project.local.protocol                 | https                                                                            |
| project.local.hostname                 | local.site.test                                                         |
| project.machine_name                   | site                                                                         |
| project.prefix                         | PRJ                                                                              |
| project.human_name                     | site.gov                                                                |
| project.profile.name                   | lightning                                                                        |
| setup.strategy                         | sync                                                                             |
| setup.dump-file                        |                                                                                  |
| setup.drupal.install.import-config     | 1                                                                                |
| sync.files                             |                                                                                  |
| sync.exclude-paths.0                   | styles                                                                           |
| sync.exclude-paths.1                   | css                                                                              |
| sync.exclude-paths.2                   | js                                                                               |
| target-hooks.frontend-setup.dir        | /Users/me/Sites/site/docroot/themes/custom/uswds_site                     |
| target-hooks.frontend-setup.command    | npm install                                                                      |
| target-hooks.frontend-build.dir        | /Users/me/Sites/site/docroot/themes/custom/uswds_site                     |
| target-hooks.frontend-build.command    | npm run build                                                                    |
| target-hooks.frontend-test.dir         | /Users/me/Sites/site/docroot                                               |
| target-hooks.frontend-test.command     |                                                                                  |
| target-hooks.pre-commit.dir            | /Users/me/Sites/site/docroot                                               |
| target-hooks.pre-commit.command        |                                                                                  |
| target-hooks.pre-config-import.dir     | /Users/me/Sites/site/docroot                                               |
| target-hooks.pre-config-import.command |                                                                                  |
| target-hooks.post-deploy-build.dir     | /Users/me/Sites/site/deploy/docroot                                        |
| target-hooks.post-deploy-build.command |                                                                                  |
| target-hooks.post-setup-build.dir      | /Users/me/Sites/site/docroot                                               |
| target-hooks.post-setup-build.command  |                                                                                  |
| target-hooks.post-deploy.dir           | /Users/me/Sites/site/docroot                                               |
| target-hooks.post-deploy.command       |                                                                                  |
| tests.run-server                       |                                                                                  |
| tests.server.port                      | 8888                                                                             |
| tests.server.url                       | http://127.0.0.1:8888                                                            |
| validate.lint.filesets.0               | files.php.custom.modules                                                         |
| validate.lint.filesets.1               | files.php.custom.themes                                                          |
| validate.lint.filesets.2               | files.php.tests                                                                  |
| validate.twig.filesets.0               | files.twig                                                                       |
| validate.yaml.filesets.0               | files.yaml                                                                       |
| vm.enable                              | 1                                                                                |
| vm.config                              | /Users/me/Sites/site/box/config.yml                                        |
| simplesamlphp                          | TRUE                                                                             |
| environment                            | local                                                                            |
| multisites.0                           | default                                                                          |
| options.decorated                      | 1                                                                                |
| options.interactive                    | 1                                                                                |
| options.progress-delay                 | 2                                                                                |
| options.simulate                       |                                                                                  |
| options.help                           |                                                                                  |
| options.quiet                          |                                                                                  |
| options.verbose                        |                                                                                  |
| options.version                        |                                                                                  |
| options.ansi                           |                                                                                  |
| options.no-ansi                        |                                                                                  |
| options.no-interaction                 |                                                                                  |
| options.yes                            |                                                                                  |
| site                                   | default                                                                          |
+----------------------------------------+----------------------------------------------------------------------------------+
```

**Comment by grasmash** [comment]
Alright, that verifies that `setup.strategy` is set and read correctly as `sync`. Seems like a bug then!

**Comment by grasmash** [comment]
This feature has been significantly refactored in 9.x and now has test coverage. 

9.0.0 was released today. Closing this issue as a fix is not planned for backport to the 8.9.x branch.

**Comment by timwood** [comment]
Thanks for the update, Matthew. We are planning on updating BLT to 9.x, Lightning to 3.1 (which includes Drupal 8.5) and DrupalVM to 4.8. Is there anything to watch out for in upgrading all this? I am aware of some Drush 9 related things, but anything else?
